### PR TITLE
Test for Brackets when starting Thimble #2509, Error cannot find module using require

### PR DESCRIPTION
**Pull request by MPierre9**
As suggested by @humphd I opened a PR for this issue which acts as a continuation of issue #2509. There seems to be an error when I do `const config = require('server/routes/config');`. The console says it can't find the module. 
require bug

The idea is to have the code test to see if the brackets server is running and output a notification. This will minimize the issue from happening to new developers (happened to me).

### CODE REVIEW

File: server/verify-brackets.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const config = require('server/routes/config');
+
+const isReachable = require('is-reachable');
+var brambleServer = config.editorHOST;
+
+console.log("Checking for Bramble Server. " + brambleServer);
+
+isReachable(brambleServer).then(reachable => {
+  
+  if(reachable)
```
Comment (humphd) [inline server/verify-brackets.js]: In JS you want to always put the `{` on the same line as the statement.  There are some weird bugs you can get into with `return` and automatic semi-colon insertion, for example:

```js
return // actually returns `undefined`
{
  data: 1
};

// vs. 

return { // does what you want
  data: 1
};
```

File: server/verify-brackets.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const config = require('server/routes/config');
+
+const isReachable = require('is-reachable');
+var brambleServer = config.editorHOST;
+
+console.log("Checking for Bramble Server. " + brambleServer);
```
Comment (humphd) [inline server/verify-brackets.js]: In the spirit of Unix, let's only print in the case of an error.  You can drop this.

File: server/verify-brackets.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const config = require('server/routes/config');
```
Comment (humphd) [inline server/verify-brackets.js]: The way that node does its module loading is that you can pass it a module name, like `is-reachable` below, or you can pass it a path, which needs to start with `.` or `..` or the like.  You probably want this:

```js
const config = require('./routes/config');
```

File: server/verify-brackets.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+isReachable(brambleServer).then(reachable => {
+  
+  if(reachable)
+  {
+    console.log("Bramble server is running.");    
+  }
+  else
+  {
+    console.log("Error: Bramble server is not running. Please end this process and run the Brackets server with: npm start");
```
Comment (humphd) [inline server/verify-brackets.js]: Let's only print in the error case:

```js
if(!reachable) {
    console.warn("...");
}
```

File: server/verify-brackets.js (hunk 1 above)
Comment (MPierre9) [reply server/verify-brackets.js]: Oh I didn't even know that could happen with JS. Good to know.

**Comment by MPierre9** [comment]
Thanks for the input and help @humphd I'll implement those changes and do another commit 👍

File: server/verify-brackets.js (hunk 3 above)
Comment (MPierre9) [reply server/verify-brackets.js]: Oh ok I was using the wrong path. I looked up the error and saw people using "./" but I was doing "./server/routes/config".

**Comment by MPierre9** [comment]
Implemented the changes @humphd mentioned. It seems the Travis CI check doesn't like me outputting to the console. Here are some screenshots of the build working. 

working brackets 1


working brackets 2

**Comment by gideonthomas** [comment]
@MPierre9, as per https://github.com/mozilla/thimble.mozilla.org/blob/master/CONTRIBUTING.md#submit-a-pull-request, run `npm run test:lint:fix` to fix your linting issues and commit the changes. That should make Travis happy.

File: scripts/verify-brackets.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const config = require("../server/routes/config");
+const isReachable = require("is-reachable");
+var brambleServer = config.editorHOST;
```
Comment (gideonthomas) [inline scripts/verify-brackets.js]: nit: this should be a `const` since we don't change its value at any point.

Comment (MPierre9) [reply scripts/verify-brackets.js]: Oh ok makes sense, Thanks!

File: scripts/verify-brackets.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const config = require("../server/routes/config");
+const isReachable = require("is-reachable");
+const brambleServer = config.editorHOST;
+const colors = require("colors");
+
+isReachable(brambleServer).then(reachable => {
+  if (!reachable) {
+    console.warn(
+      colors.yellow(
+        `Error: Bramble server is not running. Please end this process and run the Brackets server with: npm start`
```
Comment (gideonthomas) [inline scripts/verify-brackets.js]: ah darn, sorry I missed this, could you change this to:
```
`Error: Brackets is not running. In a separate terminal window, run Brackets using npm start in the brackets folder.`
```

Comment (MPierre9) [reply scripts/verify-brackets.js]: No worries, I'm on it 👍

**Comment by gideonthomas** [comment]
Awesome work @MPierre9! Thanks for sticking with it.

**Comment by MPierre9** [comment]
Thanks for all the help @gideonthomas :)

**Pull request merged**